### PR TITLE
[Actions] Bump `needs-attention` to `v2.0.0`

### PR DESCRIPTION
**Pull request by Pranav-yadav**
## Summary:

This diff bumps `hramos/needs-attention` to `v2.0.0`
`v2.0.0` is a `latest` **BREAKING** release with a number of improvements made by @Pranav-yadav i.e. ***me*** 😎

P.S. Just to have node16 across RN workflows I did [that work](https://github.com/hramos/needs-attention/commits/?author=Pranav-yadav) 🤣 and released the v2 :)

### Major Improvements:
- Bump `Node.js` to `v16`
- Bump `actions/checkout` to `v3`
- Add **TypeScript**
- Generate and publish **fully compiled** dist (with deps) instead of `node_modules`

### Ref.:
- Full Changelog for `v2.0.0`: https://github.com/hramos/needs-attention/releases/tag/v2.0.0 
- Marketplace `v2.0.0`: https://github.com/marketplace/actions/needs-attention?version=v2.0.0

## Changelog:

[GENERAL] [SECURITY] - Bump `hramos/needs-attention` to `v2.0.0`

## Test Plan

- Workflow should add and remove respective labels as usual.

### CODE REVIEW

**Comment by Pranav-yadav** [comment]
ping @cortinico, is there anything that needs to be done from my side?
Rebased :)

**Comment by cortinico** [comment]
> ping @cortinico, is there anything that needs to be done from my side?

yup, we'd like to move `needs-attention` to `react-native-community` before merging this

**Comment by Pranav-yadav** [comment]
> > ping @cortinico, is there anything that needs to be done from my side?
> 
> yup, we'd like to move `needs-attention` to `react-native-community` before merging this

Yeah, that would be great.
But, I'm not the owner, **Héctor Ramos** owns the repo.
You may ping him.

**Comment by cortinico** [comment]
I've messaged him and will get back to you once we can unblock this

File: .github/workflows/needs-attention.yml
Comment (cortinico) [inline .github/workflows/needs-attention.yml:21]:
```suggestion
        uses: react-native-community/needs-attention@v2.0.0
```

**Comment by cortinico** [comment]
@Pranav-yadav are you still interested in merging this? If so, we'll need to rebase + apply the suggestion I added

**Review by cortinico — APPROVED** [review]

**Pull request closed without merging**
